packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -4,14 +4,81 @@ const chalk = require(`chalk`)
 const { formatPluginOptionsForCLI } = require(`./plugin-options`)
 const { CODES } = require(`./report`)
 
+/**
+ * Generate a user friendly error message.
+ *
+ * Contentful's API has its own error message structure, which might change depending of internal server or authentification errors.
+ *
+ * Additionally the SDK strips the error object, sometimes:
+ * https://github.com/contentful/contentful.js/blob/b67b77ac8c919c4ec39203f8cac2043854ab0014/lib/create-contentful-api.js#L89-L99
+ *
+ * This code tries to work around this.
+ */
+const createContentfulErrorMessage = e => {
+  // If we got a response, it is very likely that it is a Contentful API error.
+  if (e.response) {
+    let parsedContentfulErrorData = {}
+
+    // Parse JSON response data, and add it to the object.
+    if (typeof e.response.data === `string`) {
+      try {
+        parsedContentfulErrorData = JSON.parse(e.response.data)
+        if (typeof parsedContentfulErrorData !== `object`) {
+          throw new Error()
+        }
+      } catch (err) {
+        e.message = `Unable to extract API error data from:\n${e.response.data}`
+      }
+      // If response data was parsed alredy, just add it.
+    } else if (typeof e.response.data === `object`) {
+      parsedContentfulErrorData = e.response.data
+    }
+
+    e = { ...e, ...e.response, ...parsedContentfulErrorData }
+  }
+
+  let errorMessage = [
+    // Generic error values
+    e.code && `${e.code}`,
+    e.status && `${e.status}`,
+    e.statusText,
+    // Contentful API error response values
+    e?.sys?.id,
```

Comment:
```suggestion
    e.sys?.id,
```

Reply:
we can not be sure if sys is available, for example network errors won't contain that value